File: tests/fixtures/sample.rb
require 'json'
require 'net/http'

module UserManagement
  class User
    attr_accessor :id, :name, :email

    def initialize(id:, name:, email:)
      @id = id
      @name = name
      @email = email
    end

    def display_name
      "#{name} <#{email}>"
    end
  end

  class UserRepository
    def find_by_id(id)
      # lookup
    end

    def create(attrs)
      User.new(**attrs)
    end
  end
end

def validate_email(email)
  email.match?(/\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i)
end

File: tests/fixtures/sample.go
package main

import (
	"fmt"
	"net/http"
)

type User struct {
	ID    int    `json:"id"`
	Name  string `json:"name"`
	Email string `json:"email"`
}

type UserRepository interface {
	FindByID(id int) (*User, error)
	FindAll() ([]*User, error)
	Create(user *User) error
}

type UserService struct {
	repo UserRepository
}

func NewUserService(repo UserRepository) *UserService {
	return &UserService{repo: repo}
}

func (s *UserService) GetUser(id int) (*User, error) {
	return s.repo.FindByID(id)
}

func HandleGetUser(w http.ResponseWriter, r *http.Request) {
	fmt.Fprintf(w, "Hello")
}

func main() {
	http.HandleFunc("/users", HandleGetUser)
	http.ListenAndServe(":8080", nil)
}
